### DISCUSSION TITLE
Apple profile configuration for ServerCertificateIssuerCommonName

### DISCUSSION THREAD

**Discussion by TomHsiung** [strongswan/strongswan#1559] [Q&A]
Hello,

I plan to get certificate-key pairs from Let's Encrypt. There are four files including cert.pem, fullchain.pem, chain.pem, and privkey.pem.

When I come to the step of profile making via Apple Configurator, I don't know what should be the content of option ServerCertificateIssuerCommonName. Definitely, what I got from Let's Encrypt is not root CA, and the chain.pem actually contains two intermediate CAs. Which one should I fill? Thanks.

**Comment by tobiasbrunner** [comment]
I don't know if Apple expects the CN of the immediate issuer of the server certificate, or if they allow configuring even the root CA. From the name of the setting it sounds more like the former than the latter. However, they might not have considered multi-level PKIs when they decided on that name, because the documentation says:

> If set, this field will cause IKE to send a certificate request based on this certificate issuer to the server.

And that's only possible if the certificate that matches that CN is available locally to build a hash that can be sent, which is usually not the case with intermediate CA certificates (although, I guess, you could embed them all in the profile). 

Anyway, if it's the immediate issuer, it would be `R3` (the complete subject DN of that certificate is `C=US, O=Let's Encrypt, CN=R3`), and if it's the root CA then it would be `DST Root CA X3` (the complete DN is `O=Digital Signature Trust Co., CN=DST Root CA X3`).

**Reply by TomHsiung** [reply]
`authentication of 'domain.com' (myself) with ECDSA-384 signature successful`

However, the VPN tunnel failed to establish. I use eap-mschapv2 method for my clients. Does it require me to add the certificate of the issuer to the Apple profile? I have not done so.

**Reply by tobiasbrunner** [reply]
Check the log. In particular, if the client sends a certificate request (that's what the `ServerCertificateIssuerCommonName` setting should trigger). If you've made the intermediate CA certificates available to the daemon, they should get sent in the response along with the server certificate. If the client does not send a certificate request, you can force the sending of intermediate CA certificates (`send_cert=always` in swanctl.conf), but I guess that would indicate that the `ServerCertificateIssuerCommonName` setting is incorrect (so the matching to the certificate chain could still fail) or that the certificate it required to send the certificate request was not found (as mentioned above).

**Reply by TomHsiung** [reply]
Finally, it works. Summarize important configuration below.

1.The `ServerCertificateIssuerCommonName` should be R3.
2.The `chain.pem` derived from Let's Encrypt should added into the profile via Apple Configrator. The `chain.pem` file actually contains two intermediate CAs (rather than the root CA). However, Apple Configurator does not recognize the file type of pem, so you need a further simple step to overcome this issue: Just rename the `chain.pem` to `chain.cer`.
